gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java

```diff
@@ -412,6 +412,12 @@
   public static final HadoopConfigurationProperty<RedactedString> GCS_ENCRYPTION_KEY_HASH =
       new HadoopConfigurationProperty<>("fs.gs.encryption.key.hash");
 
+  /** Configuration key for ignoring generation mismatch error for overwrite mutations. */
+  public static final HadoopConfigurationProperty<Boolean>
+      GCS_OVERWRITE_MUTATIONS_IGNORE_GENERATION_MISMATCH =
+          new HadoopConfigurationProperty<>(
+              "fs.gs.overwrite.mutations.ignore.generation.mismatch", false);
```

Comment:
`mutations` seems redundant - `overwrite` already assumes `mutation`. Maybe let's use shorter `fs.gs.overwrite.generation.mismatch.ignore`? (`ignore` at the end is consistent with `enable` at the end pattern that we have for other boolean properties)